client/src/main/scala/drt/client/services/handlers/ForecastFileUploadHandler.scala

```diff
@@ -0,0 +1,72 @@
+package drt.client.services.handlers
+
+import diode._
+import diode.data.{Pot, Ready}
+import drt.client.actions.Actions._
+import drt.client.components.FileUploadState
+import drt.client.logger.log
+import org.scalajs.dom
+import org.scalajs.dom.FormData
+import org.scalajs.dom.ext.AjaxException
+import upickle.default._
+
+import scala.concurrent.{Future, Promise}
+import scala.language.postfixOps
+import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
+
+case class ResponseMessage(message: String)
+
+object ResponseMessage {
+
+  import upickle.default.{macroRW, ReadWriter => RW}
+
+  implicit val rw: RW[ResponseMessage] = macroRW
+}
+
+class ForecastFileUploadHandler[M](modelRW: ModelRW[M, Pot[FileUploadState]]) extends LoggingActionHandler(modelRW) {
+
+  protected def handle: PartialFunction[Any, ActionResult[M]] = {
+
+    case ResetFileUpload() =>
+      updated(Ready(FileUploadState(state = "", message = "")))
+
+
+    case FileUploadStatus(fileUploadState: FileUploadState) =>
+      updated(Ready(fileUploadState))
+
+
+    case ForecastFileUploadAction(portCode: String, formData: FormData) =>
+
+      val tReq = new dom.XMLHttpRequest()
+      val tPromise = Promise[dom.XMLHttpRequest]()
```

Comment:
I'm not sure what the 't' in tReq and tPromise means here. Could we improve the names?

Reply:
sure 👍